sources/main.cpp

```diff
@@ -207,6 +207,27 @@ int main(int argc, char* argv[]) {
     ButtonHandler* buttonHandler = new ButtonHandler(hwFactory->getInterruptHandler(), yioapi);
     qmlRegisterSingletonType<ButtonHandler>("ButtonHandler", 1, 0, "ButtonHandler", &ButtonHandler::getQMLInstance);
 
+    // reset combination was pressed on startup
+    if (buttonHandler->resetButtonsPressed()) {
+        // create marker file
+        QFile file("/firstrun");
+        if (file.open(QIODevice::WriteOnly)) {
+            file.close();
+        }
+
+        // restore default config
+        if (QFile::exists("/boot/config.json")) {
+            QFile::remove("/boot/config.json");
+        }
+        QFile::copy("/opt/yio/app/config.json", "/boot/config.json");
+
+        // reboot the remote
+        Launcher* l = new Launcher();
```

Comment:
Memory leak!
No need for new, just `Launcher().launch("reboot");`
Btw: there will soon be a an enhanced SystemService providing reboot & shutdown to avoid using Launcher.